pkg/capabilities/capabilities.go

```diff
@@ -31,28 +31,27 @@ const (
 )
 
 type Capabilities struct {
-	have   *cap.Set
-	all    map[cap.Value]map[ringType]bool
-	bypass bool
-	lock   *sync.Mutex // big lock to guarantee all threads are on the same ring
+	have        *cap.Set
+	all         map[cap.Value]map[ringType]bool
+	bypass      bool
+	initialized bool
+	lock        *sync.Mutex // big lock to guarantee all threads are on the same ring
 }
 
-func NewCapabilities(bypass bool) (*Capabilities, error) {
-	c := &Capabilities{}
-	err := c.initialize(bypass)
-	return c, err
+func NewCapabilities(bypass bool) error {
+	Caps = Capabilities{}
```

Comment:
Doesn't this break the idea of it being a singleton? (I think this will create a new instance of Capabilities every time NewCapabilities is called, and after the first time, it won't be initialized at all.)

Also, you can use `sync.Once` instead of having the initialized bool.
Something like:

```
var once sync.Once
var caps *Capabilities // singleton for all packages

func NewCapabilities(bypass bool) (*Capabilities, error) {
        var err error
	once.Do(func() {
                caps = &Capabilities{}
                err = caps.initialize(bypass)
	})

	return caps, err
}
```

Reply:
> Doesn't this break the idea of it being a singleton?

Clearly I missed returning the reference. Fixing.

Reply:
Being fixed at https://github.com/aquasecurity/tracee/pull/2298